language/tools/vm-genesis/src/genesis_context.rs

```diff
@@ -218,7 +231,14 @@ impl<'txn> DataStore for GenesisDataCache<'txn> {
         self.data_store.move_resource_from(ap, ty)
     }
 
+    // Check first if a module reference is in the modules provided at initialization,
+    // otherwise check in the "live" data store.
+    // Normally the only code requested is in the `code_map`, it was initialized
+    // at startup.
     fn load_module(&self, module: &ModuleId) -> VMResult<Vec<u8>> {
```

Comment:
this is not the most consistent usage of the data cache.
`load_module` is the only API changed, `exists_module` and `publish_module` ignore the `code_map` entirely.
Given usage of the other API and execution in genesis this is all good, but something to keep in mind.
I don't think is worth to sweat over the risks here, but genesis usage of the VM could be reviewed a bit more.
Right now (and unrelated to this change, it was always there) code is published directly in the data store which means it's not verified by the VM (as a publish call would do). We verify the module upfront but I don't know....
so bottom line, this seems good to me
@runtian-zhou @tzakian @sblackshear let me know if you have thoughts or concerns

Reply:
I agree with your sentiment that the genesis usage is fine, but that it might be worthwhile taking a look at it and seeing if anything can be tightened up there.